main.go

```diff
@@ -40,22 +44,60 @@ func main() {
 		"Determines the minimum frequency at which watched resources are reconciled")
 	reloaderRunPeriod := flag.Duration("reloader_run_period", defaultReloaderRunPeriod,
 		"Determines the minimum frequency at which watched resources are reloaded")
-	logLevel := flag.String("log_level", "info", "Log level (debug, info, warn, error, fatal, panic).")
+	logLevel := flag.String("log_level", "info", "Log level (debug, info, warn, error).")
+	enableJSONLog := flag.Bool("enable_json_log", false, "Enable JSON logging")
```

Comment:
I am okay with this, but I am not a big fan of underscored flags. I would create a new issue to rewrite these args to use dashes instead of underscores (`collector-sync-period` etc). 

Apart from that, I'd just rename the `enable_json_log` to `log_format` with allowed `text` and `json` values in this PR, and do other flag-related changes in subsequent PR.